tests/ui/drop/lint-if-let-rescope-gated.rs

```diff
@@ -0,0 +1,30 @@
+// This test checks that the lint `if_let_rescope` does not action if either
+// - Edition 2024 is not enabled, or
+// - the feature gate `if_let_rescope` is disabled
+
+//@ revisions: neither no_feature_gate
+//@ check-pass
+//@ [neither] edition: 2021
```

Comment:
Suggestion [REV8]: I don't think edition matters here, but instead we only want to check feature gate, i.e. I think we can drop `neither`, but instead have 2 revisions:

1. `yes_feature_gate`: `#[cfg_attr(yes_feature_gate, feature(if_let_rescope))]` + `check-pass`
2. `no_feature_gate`: fails

As you see fit.

Reply:
I would still keep `edition: 2021` because the migration will happen from 2021 to 2024. Without `edition: 2021` it is Edition 2015 by default.

Reply:
That's fine yeah.

Reply:
Okay!
Applied